edb/server/ha/passive.py

```diff
@@ -0,0 +1,121 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2021-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import *
+
+import asyncio
+import enum
+import logging
+
+from . import base
+
+
+UNHEALTHY_MIN_TIME = 30
+UNEXPECTED_DISCONNECTS_THRESHOLD = 0.6
+
+logger = logging.getLogger("edb.pgcluster")
+
+
+class State(enum.Enum):
+    HEALTHY = 1
+    UNHEALTHY = 2
+    FAILOVER = 3
+
+
+class PassiveHASupport:
+    _state: State
+    _unhealthy_timer_handle: Optional[asyncio.TimerHandle]
```

Comment:
I like how you encapsulated the state machine in this class, but can you also add a docstring summarizing it? (You can copy the relevant bits from the GH discussion)